src/Stratis.Bitcoin.Features.BlockStore/BlockStoreBehavior.cs

```diff
@@ -131,49 +141,180 @@ protected override void DetachCore()
             this.logger.LogTrace("(-)");
         }
 
-        private Task ProcessMessageAsync(NetworkPeer node, IncomingMessage message)
+        private async Task ProcessMessageAsync(NetworkPeer peer, IncomingMessage message)
+        {
+            this.logger.LogTrace("({0}:'{1}',{2}:'{3}')", nameof(peer), peer.RemoteSocketEndpoint, nameof(message), message.Message.Command);
+
+            switch (message.Message.Payload)
+            {
+                case GetDataPayload getDataPayload:
+                    if (!this.CanRespondToGetDataPayload)
+                    {
+                        this.logger.LogTrace("Can't respond to 'getdata'.");
+                        break;
+                    }
+
+                    await this.ProcessGetDataAsync(peer, getDataPayload).ConfigureAwait(false);
+                    break;
+
+                case GetBlocksPayload getBlocksPayload:
+                    // TODO: this is not used in core anymore consider deleting it
+                    // However, this is required for StratisX to be able to sync from us.
+
+                    if (!this.CanRespondToGetDataPayload)
```

Comment:
This should be `CanRespondToGetBlocksPayload`

Reply:
of course! thanks